### ISSUE TITLE
About the experimental results

### ISSUE THREAD

**Issue by Loshhhhh** [thuml/iTransformer#30]
Hello，May I ask why the results of the comparison model PatchTST in this paper differ so much from those in the paper PatchTST?

Experimental results from iTransformer

![itransformer](https://github.com/thuml/iTransformer/assets/93030275/00e367e2-4b1e-464e-a240-ff4cce13269f)

Experimental results from PatchTST
![patchTST](https://github.com/thuml/iTransformer/assets/93030275/3e71eddc-fcf8-4edf-a21c-e4b3c5c18490)

**Comment by WenWeiTHU** [comment]
Hello, thanks for your concen!

We carefully read through the forecasting settings of the PatchTST paper and its official [repo](https://github.com/yuqinie98/PatchTST/tree/main). There's a lot of difference here:

* **Enlarged lookback window**: PatchTST paper adopts tunable lookback length in {336, 512}, while ours adopts the unified length of 96, following the unified long-term forecasting protocol of [TimesNet](https://github.com/thuml/Time-Series-Library).
* **More epochs to train**: PatchTST paper trains the PatchTST model with 100 epochs, while we train all models with 10 epochs.
* **Learning rate**: PatchTST paper adapts a carefully designed learning rate strategy.

**Comment by Loshhhhh** [comment]
@WenWeiTHU 
    Thank you for your reply, but I still have some questions.

Question 1：
    In Appendix E2 of the iTransformer paper, all experimental results are done following the setting of TimesNet. So I checked the link [https://github.com/thuml/Time-Series-Library](url) PatchTST code，I noticed that you don't use learnable additive position encoding in your Embed python file。
    However, in the paper PatchTST, the Transformer Encoder section 3.1 mentions the use of learnable additive position encoding.
[https://github.com/yuqinie98/PatchTST/blob/main/PatchTST_supervised/layers/PatchTST_layers.py](url) The positional_encoding function in this file does the same。
    I don't know why this has changed, can you tell me? Please point out if I'm wrong. Thank you very much.


 Question 2：
    Why should the epoch be turned down? What factors are taken into account?

**Comment by WenWeiTHU** [comment]
Thanks very much for your rigorousness. 

For Q1, it has more to do with these related issues of Time-Series-Library: https://github.com/thuml/Time-Series-Library/issues?q=patchtst. 

For Q2. As for the training protocols. We do not cut down the epochs. Under the unified learning rate decay strategy of TimesNet. The models all come to convergence before 10 epoch under the provided benchmarks.

<img width="1186" alt="image" src="https://github.com/thuml/iTransformer/assets/39073236/ae17752c-918b-42ad-9828-cc6bd44d02b7">

In fact, the training epochs for multivariate forecasting benchmarks have not been overly scaled up since PatchTST (Please refer to [Informer](https://github.com/zhouhaoyi/Informer2020), [Autoformer](https://github.com/thuml/Autoformer)). It can be unrealistic to elaboratively tune the learning strategies for each model, especially for models with channel indepence (CI) which is time-comsuming on benchmarks with numerous variables. We follow the unified training protocal just for the fair comparison and avoid overtuning, Please understand that we focus on the models design instead of the engineering of tuning.

**Comment by Loshhhhh** [comment]
@WenWeiTHU 
Thank you very much for your detailed explanation, but I still haven't found a suitable explanation for Q1. I think your code ignores learnable additive position encoding. If I miss it, please let me know, thank you!
